static/app/views/dashboards/widgetBuilder/hooks/useWidgetBuilderState.tsx

```diff
@@ -0,0 +1,42 @@
+import {useCallback, useMemo} from 'react';
+
+import {useTitle} from 'sentry/views/dashboards/widgetBuilder/hooks/useTitle';
+
+export const BuilderStateAction = {
+  SET_TITLE: 'SET_TITLE',
+} as const;
+
+type WidgetAction = {payload: string; type: typeof BuilderStateAction.SET_TITLE};
+
+interface WidgetBuilderState {
+  title?: string;
+}
+
+function useWidgetBuilderState(): {
```

Comment:
I don't know exactly how to test this yet, but I think it may make more sense to test any results of "X field was updated, so Y field was also updated" relationships. Since I don't have any of those yet I didn't create a test file.

Reply:
I made better updates in #81412, so actually you don't need to review this one so closely